### PR TITLE
Add heuristic func to set plugin workspace base for windows containers correctly

### PR DESCRIPTION
**Pull request by 6543**
regression of  #3933

address one concern of  #4274

### CODE REVIEW

**Comment by 6543** [comment]
@cduchenoy you might verify that this works for you! :)

File: pipeline/frontend/yaml/compiler/convert.go [hunk 1]
```diff
@@ -205,7 +228,11 @@ func (c *Compiler) stepWorkingDir(container *yaml_types.Container) string {
 	}
 	base := c.workspaceBase
 	if container.IsPlugin() {
-		base = pluginWorkspaceBase
+		if c.checkRunOnWindowsHeuristics() {
```
Comment (6543) [inline pipeline/frontend/yaml/compiler/convert.go:231]:
```suggestion
		// plugins have a predefined workspace base to not tamper with entrypoint executables
		if c.checkRunOnWindowsHeuristics() {
```

**Comment by cduchenoy** [comment]
Hi,

@6543

Thank you so much !

To try it I need to build the Woodpecker server image with this patch using <https://woodpecker-ci.org/docs/development/getting-started>
is that it?

As a reminder, we are not devs but infrastructure people

**Comment by pat-s** [comment]
You can use the tag `pull_4286`.

**Comment by 6543** [comment]
> As a reminder, we are not devs but infrastructure people

@cduchenoy that's what `build_pr_image` lable is for :)

server [image](https://hub.docker.com/layers/woodpeckerci/woodpecker-server/pull_4286/images/sha256-0c263f35c8df8fab7b399711b7b7544ab4eb743abfacc3b22db5b5fd3fd7096b?context=explore)
agent [image](https://hub.docker.com/layers/woodpeckerci/woodpecker-agent/pull_4286/images/sha256-2a89f3d2150308555ac504c12254fdb85404ec413d7d07e41b2d5efdf32c519e?context=explore)

**Comment by 6543** [comment]
@cduchenoy did you had time to test it?

**Comment by cduchenoy** [comment]
Hi,

Sorry but I haven't been able to test yet

It is in production at our customer and testing is complicated

We are in the process of migrating our old gitea/drone stack to the forgejo/woodpecker couple

Once the infrastructure is finalized (well advanced :)) we will be able to test because our infrastructure and no longer that of a customer

We are also preparing the expected PRs and a teams notification plugin

**Comment by cduchenoy** [comment]
Hi,

I was finally able to test it works!

I had to create the agent for Windows with the patch because I had a grpc version mismatch error

The Windows docker image windows-agent can be found on docker-hub:

```bash
docker pull gecoit84/woodpecker-agent:pull_4286
```

Do you think you can apply the same fix for the steps command?

Indeed, for the pipeline to work you must add the parameter 

```yaml
...
workspace:
  base: C:\tmp
```

Without that

image

**Comment by 6543** [comment]
I just had an idea witch would not require an heuristic: why not prepend `C:\` directly in the docker backend if the backend itselve detects windows as an host ... :thinking:

**Comment by cduchenoy** [comment]
Yes it's the solution :+1:

**Comment by 6543** [comment]
@cduchenoy ok you might have a look at #4381 ... through I have windows container now running and might do actual testing with woodpecker on my end too now :)

**Pull request closed without merging**